core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -764,7 +557,7 @@ public void stop() throws Exception {
     server.stop();
   }
 
-  private void doShutdown() {
+  void doShutdown() {
```

Comment:
As long as we have resource extensionsI think we have to contend with at least some lifecycle management. Package-protected in conjunction with final seems like a decent compromise for v1. In the long run I question if resource extensions are still needed since we can add multiple applications to a server now.